### PR TITLE
r/lakeformation_permissions: Fix update bug

### LINKED ISSUE #19397: bug error aws_data_lake_permission for coloumn
i am trying to provide the permission for column level, but it is showing bug error.

### LINKED ISSUE #19409: aws_lakeformation_permissions column level permission is not working
aws_lakeformation_permissions column level permission is not working  even i tried wild card value.

resource "aws_lakeformation_permissions" "grantpermission-columns-users_wildcard" {
  count                         = var.columns_wildcard ? 1 : 0
  principal                     = var.user_principal
  permissions                   = var.user_permissions_coloumn
  table_with_columns {
    database_name         = var.user_database
    name                  = var.user_database_table
    excluded_column_names = var.excluded_column_names
    wildcard              = var.columns_wildcard
  }
}

  user_permissions_table       = ["CREATE_TABLE", "ALTER", "DROP"]
  user_permissions_coloumn     = ["CREATE_TABLE", "ALTER", "DROP"]
  columns_wildcard             = true
  excluded_column_names        = ["policyid"] ## must and only when columns_wildcard arguement is enabled
  user_principal               = "arn:aws:iam:::role/LakeFormationAnalyst"
  athena_query_savings         = "supertest-dataprocessed"
  crawler_data_source          = "supertest-datasource"

### LINKED ISSUE #19465: Lakeformation table permissions need to be alphabetically ordered - otherwise aws_lakeformation_permissions stay out of sync
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

3.42

* aws_lakeformation_permissions

### Terraform Configuration Files

```hcl

resource "aws_lakeformation_permissions" "db_perm" {
  count       = var.is_ext == false ? 1 : 0
  principal   = var.read_group
  permissions = ["SELECT","DESCRIBE" ]


  table {
    database_name       = "${var.db_name}_db"
    wildcard = true
  }
}

```

### Expected Behavior

replanning after apply doesnt show changes

### Actual Behavior

 module.database["grp2"].aws_lakeformation_permissions.db_perm[0] must be replaced
-/+ resource "aws_lakeformation_permissions" "db_perm" {
      ~ id                            = "466990021" -> (known after apply)
      ~ permissions                   = [ # forces replacement
          - "DESCRIBE",
            "SELECT",
          + "DESCRIBE",
        ]
      ~ permissions_with_grant_option = [] -> (known after apply)

### Steps to Reproduce

apply above resource and plan again

### LINKED ISSUE #19643: aws_lakeformation_permissions inconsistent result after apply
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.15.5
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.43.0
```

### Affected Resource(s)
* aws_lakeformation_permissions

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_lakeformation_permissions" "data_producer_database" {

  provider = aws.lf_data_producer

  principal = var.org_arn

  permissions                   = ["DESCRIBE"]
  permissions_with_grant_option = ["DESCRIBE"]
  
  database {
    name       = aws_glue_catalog_database.data_producer.name
    catalog_id = data.aws_caller_identity.lf_data_producer.account_id
  }
}

variable "org_arn" {
  default = "arn:aws:organizations::xxx:organization/o-zzzzzz"
}

```

### Expected Behavior
Permissions to be added to the Glue database that gives the entire AWS Organization `DESCRIBE` access and the changes to be reflected in the state file.

### Actual Behavior
The changes are applied in the AWS, but the state does not reflect the changes. This only occurs when applying permissions to organizations and organizational units, when giving access to other accounts, this error does not occur.

Subsequent applies creates another resource in AWS and give the same error, resulting in multiple permission entries for the same Org/OU.

```bash
aws_lakeformation_permissions.data_producer_database: Creating...
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to aws_lakeformation_permissions.data_producer_database, provider
│ "provider[\"registry.terraform.io/hashicorp/aws\"].lf_data_producer" produced an
│ unexpected new value: Root resource was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue
│ tracker.
```

### Steps to Reproduce

1. `terraform apply`

### LINKED ISSUE #19860: r/lakeformation_permissions: Add unit tests for permission filtering
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Relates #19817 

The file `aws/internal/lakeformation/filter.go` needs associated unit tests. These will be medium complexity because we'll need varying lists of "allPermissions", various inputs (`ListPermissionsInput`) plus columns, exclude columns, and wildcards, and then to verify that we get the right results.

### New or Affected Resource(s)

* aws_lakeformation_permissions

### References

* #19817

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #19643
Closes #19465
Closes #19397
Closes #19409
Closes #19860

This PR address 3 problems:
1. The order of `column_names` or `exclude_column_names` could cause diffs when reapplying (see #19465)
2. Updating twc config caused an inconsistent result error (see #19643)
3. The resource previously may not have removed deleted resources from state (see #13527)

Output from acceptance testing (GovCloud):

```
--- PASS: TestAccAWSLakeFormation_serial (621.11s)
    --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings (62.28s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource (17.66s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/basic (14.84s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/disappears (15.05s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId (14.73s)
    --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions (160.16s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/database (23.30s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/disappears (83.41s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/basic (27.11s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/dataLocation (26.34s)
    --- PASS: TestAccAWSLakeFormation_serial/TablePermissions (118.90s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableName (23.88s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableWildcard (23.10s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableWildcardPermissions (23.63s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/implicitTablePermissions (24.34s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/selectPermissions (23.95s)
    --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions (144.08s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/tableWithColumns (66.52s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardExcludedColumnsPermissions (25.02s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardPermissions (28.10s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/implicitTableWithColumnsPermissions (24.43s)
    --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions (135.70s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/databaseDataSource (26.72s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/tableDataSource (27.41s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/tableWithColumnsDataSource (24.83s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/basicDataSource (27.93s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/dataLocationDataSource (28.80s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	622.403s
```

### CODE REVIEW

File: aws/resource_aws_lakeformation_permissions.go [hunk 1]
```diff
@@ -405,18 +425,33 @@ func resourceAwsLakeFormationPermissionsRead(d *schema.ResourceData, meta interf
 	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, lakeformation.ErrCodeEntityNotFoundException) {
 		log.Printf("[WARN] Resource Lake Formation permissions (%s) not found, removing from state", d.Id())
 		d.SetId("")
+
+		if true {
+			return fmt.Errorf("death note: %d", 1)
+		}
```
Comment (ewbankkit) [inline aws/resource_aws_lakeformation_permissions.go:431]: Should probably remove this 😄.

File: aws/resource_aws_lakeformation_permissions.go [hunk 2]
```diff
@@ -405,18 +425,33 @@ func resourceAwsLakeFormationPermissionsRead(d *schema.ResourceData, meta interf
...
 		return nil
 	}
 
 	if !d.IsNewResource() && len(allPermissions) == 0 {
 		log.Printf("[WARN] Resource Lake Formation permissions (%s) not found, removing from state (0 permissions)", d.Id())
 		d.SetId("")
+
+		if true {
+			return fmt.Errorf("death note: %d", 3)
+		}
```
Comment (ewbankkit) [inline aws/resource_aws_lakeformation_permissions.go:453]: Should probably remove this 😄.

File: aws/resource_aws_lakeformation_permissions.go [hunk 3]
```diff
@@ -405,18 +425,33 @@ func resourceAwsLakeFormationPermissionsRead(d *schema.ResourceData, meta interf
...
 		return nil
 	}
 
 	if !d.IsNewResource() && tfawserr.ErrMessageContains(err, "AccessDeniedException", "Resource does not exist") {
 		log.Printf("[WARN] Resource Lake Formation permissions (%s) not found, removing from state: %s", d.Id(), err)
 		d.SetId("")
+
+		if true {
+			return fmt.Errorf("death note: %d", 2)
+		}
```
Comment (ewbankkit) [inline aws/resource_aws_lakeformation_permissions.go:442]: Should probably remove this 😄.

File: aws/resource_aws_lakeformation_permissions.go [hunk 4]
```diff
@@ -449,18 +484,38 @@ func resourceAwsLakeFormationPermissionsRead(d *schema.ResourceData, meta interf
...
 		)
 	}
 
 	if len(cleanPermissions) == 0 {
 		log.Printf("[WARN] Resource Lake Formation permissions (%s) not found, removing from state", d.Id())
 		d.SetId("")
+
+		if true {
+			return fmt.Errorf("death note: %d, %v", 4, allPermissions)
+		}
```
Comment (ewbankkit) [inline aws/resource_aws_lakeformation_permissions.go:517]: Should probably remove this 😄.

File: aws/resource_aws_lakeformation_permissions.go (hunk 1 above)
Comment (YakDriver) [reply aws/resource_aws_lakeformation_permissions.go]: Early version

File: aws/internal/service/lakeformation/filter.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
+package lakeformation
```
Comment (ewbankkit) [inline aws/internal/service/lakeformation/filter.go:1]: Is it possible to add unit tests for the functions in this file?

Comment (YakDriver) [reply aws/internal/service/lakeformation/filter.go:1]: Potentially.

Comment (ewbankkit) [reply aws/internal/service/lakeformation/filter.go:1]: It would definitely help with long-term maintainability.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSLakeFormation_serial'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSLakeFormation_serial -timeout 180m
=== RUN   TestAccAWSLakeFormation_serial
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/basic
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/disappears
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions/basic
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions/dataLocation        
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions/database
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions/disappears
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/implicitTablePermissions
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/selectPermissions
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/tableName
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/tableWildcard
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/tableWildcardPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardExcludedColumnsPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/implicitTableWithColumnsPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/tableWithColumns
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/tableDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/tableWithColumnsDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/basicDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/dataLocationDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/databaseDataSource
--- PASS: TestAccAWSLakeFormation_serial (605.13s)
    --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings (44.50s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/basic (12.02s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/disappears (10.82s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId (10.51s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource (11.15s)
    --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions (159.32s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/basic (21.95s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/dataLocation (27.88s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/database (23.02s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/disappears (86.48s)
    --- PASS: TestAccAWSLakeFormation_serial/TablePermissions (123.23s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/implicitTablePermissions (24.93s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/selectPermissions (23.23s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableName (26.52s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableWildcard (22.68s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableWildcardPermissions (25.87s)
    --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions (139.62s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardExcludedColumnsPermissions (24.28s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardPermissions (26.70s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/implicitTableWithColumnsPermissions (25.95s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/tableWithColumns (62.68s)
    --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions (138.46s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/tableDataSource (30.85s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/tableWithColumnsDataSource (28.31s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/basicDataSource (22.06s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/dataLocationDataSource (29.11s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/databaseDataSource (28.13s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	609.295s
```

##### GovCloud

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSLakeFormation_serial'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSLakeFormation_serial -timeout 180m
=== RUN   TestAccAWSLakeFormation_serial
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions/basic
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions/dataLocation
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions/database
=== RUN   TestAccAWSLakeFormation_serial/BasicPermissions/disappears
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/tableWildcardPermissions
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/implicitTablePermissions
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/selectPermissions
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/tableName
=== RUN   TestAccAWSLakeFormation_serial/TablePermissions/tableWildcard
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardExcludedColumnsPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/implicitTableWithColumnsPermissions
=== RUN   TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/tableWithColumns
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/basicDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/dataLocationDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/databaseDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/tableDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataSourcePermissions/tableWithColumnsDataSource
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/basic
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/disappears
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId
=== RUN   TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource
--- PASS: TestAccAWSLakeFormation_serial (651.02s)
    --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions (166.47s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/basic (29.41s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/dataLocation (29.60s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/database (23.57s)
        --- PASS: TestAccAWSLakeFormation_serial/BasicPermissions/disappears (83.89s)
    --- PASS: TestAccAWSLakeFormation_serial/TablePermissions (135.03s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableWildcardPermissions (26.15s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/implicitTablePermissions (30.01s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/selectPermissions (25.24s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableName (24.54s)
        --- PASS: TestAccAWSLakeFormation_serial/TablePermissions/tableWildcard (29.08s)
    --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions (150.83s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardExcludedColumnsPermissions (25.07s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/columnWildcardPermissions (25.41s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/implicitTableWithColumnsPermissions (30.24s)
        --- PASS: TestAccAWSLakeFormation_serial/TableWithColumnsPermissions/tableWithColumns (70.11s)
    --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions (137.93s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/basicDataSource (24.52s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/dataLocationDataSource (30.67s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/databaseDataSource (28.41s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/tableDataSource (29.13s)
        --- PASS: TestAccAWSLakeFormation_serial/DataSourcePermissions/tableWithColumnsDataSource (25.20s)
    --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings (60.77s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/basic (15.54s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/disappears (14.82s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/withoutCatalogId (14.78s)
        --- PASS: TestAccAWSLakeFormation_serial/DataLakeSettings/dataSource (15.62s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	654.295s
```

File: aws/internal/service/lakeformation/filter.go (hunk 5 above)
Comment (YakDriver) [reply aws/internal/service/lakeformation/filter.go:1]: You're definitely right. It's just how complicated the useful tests would be. I'm going to create a tracking issue so we can get this in.

**Pull request merged**